modules/system-view/src/test/java/org/apache/ignite/internal/systemview/SystemViewTest.java

```diff
@@ -0,0 +1,255 @@
+                .dataProvider(dataProvider)
+                .build();
+
+        assertEquals("view", view.name(), "name");
+        assertEquals(2, view.columns().size(), "columns");
+
+        expectColumn(view.columns().get(0), "c1", int.class);
+        expectColumn(view.columns().get(1), "c2", Long.class);
+
+        assertSame(dataProvider, view.dataProvider(), "data provider");
+    }
+
+    /** Builds a node view. */
+    @Test
+    public void buildNodeView() {
+        Supplier<AsyncCursor<Dummy>> dataProvider = dataProvider();
+
+        NodeSystemView<Dummy> view = NodeSystemView.<Dummy>builder()
+                .name("view")
+                .addColumn("c1", int.class, (d) -> 0)
+                .addColumn("c2", Long.class, (d) -> 1L)
+                .nodeNameColumnAlias("node_name")
+                .dataProvider(dataProvider)
+                .build();
+
+        assertEquals("view", view.name(), "name");
+        assertEquals(2, view.columns().size(), "columns");
+
+        expectColumn(view.columns().get(0), "c1", int.class);
+        expectColumn(view.columns().get(1), "c2", Long.class);
+
+        assertSame(dataProvider, view.dataProvider(), "data provider");
+        assertEquals("node_name", view.nodeNameColumnAlias(), "node name column alias");
+    }
+
+    /** Reject a node view without node name alias. */
+    @Test
+    public void rejectNodeViewWithoutNodeNameColumnAlias() {
+        expectThrows(NullPointerException.class, () -> {
+            NodeSystemView.<Dummy>builder()
+                    .name("name")
+                    .addColumn("c1", int.class, (d) -> 0)
+                    .dataProvider(dataProvider())
+                    .build();
+        }, "nodeNameColumnAlias");
+    }
+
+    /**
+     * Tests for {@link NodeSystemView.Builder}.
+     */
+    @Nested
+    public class NodeViewBuilderTest extends BuilderTest<NodeSystemView<Dummy>, Builder<Dummy>> {
+
+        @Override
+        protected NodeSystemView.Builder<Dummy> newBuilder() {
+            return NodeSystemView.builder();
+        }
+    }
+
+    /**
+     * Tests for {@link ClusterSystemView.Builder}.
+     */
+    @Nested
+    public class ClusterViewBuilderTest extends BuilderTest<ClusterSystemView<Dummy>, ClusterSystemView.Builder<Dummy>> {
+
+        @Override
+        protected ClusterSystemView.Builder<Dummy> newBuilder() {
+            return ClusterSystemView.builder();
+        }
+    }
+
+    /**
+     * Common tests for view builder classes.
+     *
+     * @param <V> View type.
+     * @param <B> Builder type.
+     */
+    public abstract class BuilderTest<V extends SystemView<Dummy>,
+            B extends SystemView.SystemViewBuilder<V, Dummy, B>> {
+
+        protected abstract B newBuilder();
+
+        /** Reject a view with {@code null} name. */
+        @Test
+        public void rejectViewWithNullName() {
+            expectThrows(NullPointerException.class, () -> {
```

Comment:
a bit confused, why you decide that NPE here is a correct behavior ? Probably i miss smth.

Reply:
But I have done it, because other null checks in constructors (see Catalog objects)  do the same.

Reply:
AFAIK, we thought about adding validation to the Builder.
And throw an exception with meaningful error code (system_view_definition_err) and message.

Reply:
I reckon system views deserve an ErrorGroup of their own.

Reply:
Let's add a new ErrorGroup for system views.